pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -248,13 +252,15 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable SqlNodeAndOption
     if (!hasAccess) {
       _brokerMetrics.addMeteredGlobalValue(BrokerMeter.REQUEST_DROPPED_DUE_TO_ACCESS_ERROR, 1);
       requestContext.setErrorCode(QueryException.ACCESS_DENIED_ERROR_CODE);
+      _brokerQueryEventListener.onQueryCompletion(new BrokerQueryEventInfo(requestContext));
       throw new WebApplicationException("Permission denied", Response.Status.FORBIDDEN);
     }
 
     long requestId = _brokerIdGenerator.get();
     requestContext.setRequestId(requestId);
     JsonNode sql = request.get(Broker.Request.SQL);
     if (sql == null) {
+      _brokerQueryEventListener.onQueryCompletion(new BrokerQueryEventInfo(requestContext));
```

Comment:
We are not setting the error code in `RequestContext` in this path, so this might get logged as success?